apps/rush-lib/src/cli/actions/GenerateAction.ts

```diff
@@ -110,19 +120,29 @@ export default class GenerateAction extends BaseRushAction {
       }
     } catch (ex) {
       console.log();
-      console.log('There was a problem reading the shrinkwrap file. Proceeeding with "rush generate".');
+      console.log('There was a problem reading the shrinkwrap file. Proceeding with "rush generate".');
     }
 
     return installManager.ensureLocalPackageManager(false).then(() => {
       installManager.createTempModules(true);
 
-      // Delete both copies of the shrinkwrap file
-      if (fsx.existsSync(committedShrinkwrapFilename)) {
-        console.log(os.EOL + 'Deleting ' + committedShrinkwrapFilename);
-        fsx.unlinkSync(committedShrinkwrapFilename);
-      }
-      if (fsx.existsSync(tempShrinkwrapFilename)) {
-        fsx.unlinkSync(tempShrinkwrapFilename);
+      if (this._conservativeParameter.value) {
+        if (fsx.existsSync(committedShrinkwrapFilename)) {
+          console.log(os.EOL + 'The "--conservative" flag was provided, so preserving '
+            + committedShrinkwrapFilename);
+        } else {
+          throw new Error('The "--conservative" flag cannot be used because the shrinkwrap file is missing: '
+            + committedShrinkwrapFilename);
+        }
+
+        // Copy common\config\rush\shrinkwrap.yaml --> common\temp\shrinkwrap.yaml
+        console.log(os.EOL + 'Updating ' + tempShrinkwrapFilename);
+        fsx.copySync(committedShrinkwrapFilename, tempShrinkwrapFilename);
+      } else {
```

Comment:
>else { [](start = 8, length = 6)

not unlinking committed shrinkwrap file?

Reply:
we discussed this offline and it is ok

---
In reply to: [171738895](https://github.com/Microsoft/web-build-tools/pull/549#discussion_r171738895) [](ancestors = 171738895)

Reply:
I documented it in a separate change log.

---
In reply to: [171738956](https://github.com/Microsoft/web-build-tools/pull/549#discussion_r171738956) [](ancestors = 171738956,171738895)